src/components/cuda/cupti_profiler.c

```diff
@@ -0,0 +1,1576 @@
+        goto fn_exit;
+    }
+    char *desc = evt_rec->desc;
+    if (desc[0] == '\0') {
+        errno = retrieve_metric_details(avail_events[gpu_id].metricsContextCreateParams.pMetricsContext,
+                                      nv_name, desc, &numdep, &temp);
+        if (errno == PAPI_OK) {
+            NVPW_CUDA_RawMetricsConfig_Create_Params nvpw_metricsConfigCreateParams = {
+                .structSize = NVPW_CUDA_RawMetricsConfig_Create_Params_STRUCT_SIZE,
+                .pPriv = NULL,
+                .activityKind = NVPA_ACTIVITY_KIND_PROFILER,
+                .pChipName = avail_events[gpu_id].chip_name,
+            };
+            NVPW_CALL( NVPW_CUDA_RawMetricsConfig_CreatePtr(&nvpw_metricsConfigCreateParams), return PAPI_EMISC );
+
+            errno = check_num_passes(nvpw_metricsConfigCreateParams.pRawMetricsConfig,
+                                   numdep, temp, &passes);
+
+            NVPW_RawMetricsConfig_Destroy_Params rawMetricsConfigDestroyParams = {
+                .structSize = NVPW_RawMetricsConfig_Destroy_Params_STRUCT_SIZE,
+                .pPriv = NULL,
+                .pRawMetricsConfig = nvpw_metricsConfigCreateParams.pRawMetricsConfig,
+            };
+            NVPW_CALL(NVPW_RawMetricsConfig_DestroyPtr((NVPW_RawMetricsConfig_Destroy_Params *) &rawMetricsConfigDestroyParams), return PAPI_EMISC);
+
+            snprintf(desc + strlen(desc), PAPI_2MAX_STR_LEN - strlen(desc), " Numpass=%d", passes);
+            if (passes > 1)
+                snprintf(desc + strlen(desc), PAPI_2MAX_STR_LEN - strlen(desc), " (multi-pass not supported)");
+        }
+        papi_free(temp);
+    }
+    strcpy(description, desc);
+fn_exit:
+    return errno;
+}
+
+static void free_all_enumerated_metrics(void)
+{
+    COMPDBG("Entering.\n");
+    int gpu_id, found;
+    NVPW_MetricsContext_Destroy_Params metricsContextDestroyParams;
+    if (avail_events == NULL)
+        return;
+    for (gpu_id=0; gpu_id<num_gpus; gpu_id++) {
+        found = find_same_chipname(gpu_id);
+        if (found > -1) {
+            avail_events[gpu_id].num_metrics = 0;
+            avail_events[gpu_id].nv_metrics = NULL;
+            continue;
+        }
+        if (avail_events[gpu_id].metricsContextCreateParams.pMetricsContext) {
+            metricsContextDestroyParams = (NVPW_MetricsContext_Destroy_Params) {
+                .structSize = NVPW_MetricsContext_Destroy_Params_STRUCT_SIZE,
+                .pPriv = NULL,
+                .pMetricsContext = avail_events[gpu_id].metricsContextCreateParams.pMetricsContext,
+            };
+            NVPW_MetricsContext_DestroyPtr(&metricsContextDestroyParams);
+        }
+        if (avail_events[gpu_id].nv_metrics)
+            free_event_name_list( &(avail_events[gpu_id].nv_metrics) );
+    }
+    papi_free(avail_events);
+    avail_events = NULL;
+}
+
+// CUPTI Profiler component API functions
+int cuptip_init(const char **pdisabled_reason)
```

Comment:
Can you return the disabled reason using another function? Something like `cuptip_error_get_last`?

Reply:
`disabled_reason` is set at multiple points. How would one function accomplish this?

Reply:
disabled reason is set at init private I think. init component temporarily sets it to a delay init message.